### DISCUSSION TITLE
Cardinality timeframe - Elastalert status index

### DISCUSSION THREAD

**Discussion by markus-nclose** [jertel/elastalert2#963] [Q&A] [answered]
Hi All, 

I have a strange issue that I'm struggling to troubleshoot and hope someone has the insight for me. 

We have a cardinality rule that looks at a timeframe of 12 hours, this timeframe condition  is explicitly set in the rule.
The alert didn't fire last night and we testing to find out why. Running elastalert-test-rule fires the alert as expected. 

My question:

Is the ```start time``` and ```end time``` in the elastalert status index the timeframe the rule is looking at? Highlighted below in screenshot?

![image](https://user-images.githubusercontent.com/38457858/192448818-7be9acc3-0249-4a0b-ab7b-304a1a155a19.png)

This time difference in the screenshot is 45min and doesn't make sense and this is what is confusing to us. 

We not sure what other cause would this not to fire. The screenshot above shows 4 hits and the cardinality is set to max:1

**Comment by jertel** [comment, marked as answer]
Is your `run_every` setting set to 45 minutes? Rules fire every `run_every` interval, regardless of their timeframe. When a rule's timeframe is larger than the `run_every` interval ElastAlert 2 will track the accumulated results after each run to see if the alert should be triggered.

**Reply by markus-nclose** [reply]
the run_every is set to 8 minutes which why it was confusing.
![image](https://user-images.githubusercontent.com/38457858/192509019-12a8414f-f4e6-49e6-b41a-3a04ef7d43fa.png)

**Reply by markus-nclose** [reply]
The test rule vs production run time differences is also messing with my trust issues :(

**Reply by jertel** [reply]
You will need to review the ElastAlert 2 logs and the associated log timestamps to recreate the sequence of events.

**Reply by markus-nclose** [reply]
Thanks for the advice. Going to fine comb the logs.

**Reply by markus-nclose** [reply]
Nothing in the logs, I will replicate with a empty index with the same timing differences to see where the issue lays.

**Reply by markus-nclose** [reply]
Feedback on my test:

I ran a test and you can see the difference that on the new/separate version, it shows the actual time difference of 12 hours which is correct.
```Sep 27 22:19:47 minimdr python[909782]: INFO:elastalert:Queried rule Test Cardinality from 2022-09-27 10:19 SAST to 2022-09-27 22:19 SAST: 2 / 2 hits
Sep 27 22:19:47 minimdr python[909782]: INFO:elastalert:Alert sent to TheHive
Sep 27 22:19:47 minimdr python[909782]: INFO:elastalert:Ran Test Cardinality from 2022-09-27 10:19 SAST to 2022-09-27 22:19 SAST: 2 query hits (1 already seen), 1 matches, 1 alerts sent
Sep 27 22:19:47 minimdr python[909782]: INFO:elastalert:Test Cardinality range 43200
```

Where the above issue it shown a random 33 minute difference which isn't configured anywhere. 
``` 2022-09-26 19:46 +0200 to 2022-09-26 21:19 +0200:```

**Reply by markus-nclose** [reply]
Just adding in for completeness, we upgraded to a version where the new Elastic schema for the meta index was used (version 7) and the problem was resolved.
